src/r1cs/mod.rs

```diff
@@ -397,13 +414,24 @@ impl<E: Engine> R1CSShape<E> {
     ck: &CommitmentKey<E>,
   ) -> Result<(RelaxedR1CSInstance<E>, RelaxedR1CSWitness<E>), NovaError> {
     // sample Z = (W, u, X)
+    #[cfg(feature = "std")]
     let Z = (0..self.num_vars + self.num_io + 1)
       .into_par_iter()
       .map(|_| E::Scalar::random(&mut OsRng))
       .collect::<Vec<E::Scalar>>();
+    #[cfg(not(feature = "std"))]
+    let Z = (0..self.num_vars + self.num_io + 1)
+      .map(|_| E::Scalar::random(&mut ChaCha20Rng::seed_from_u64(0xDEADBEEF)))
+      .collect::<Vec<E::Scalar>>();
 
+    #[cfg(feature = "std")]
     let r_W = E::Scalar::random(&mut OsRng);
+    #[cfg(feature = "std")]
     let r_E = E::Scalar::random(&mut OsRng);
+    #[cfg(not(feature = "std"))]
+    let r_W = E::Scalar::random(&mut ChaCha20Rng::seed_from_u64(0xDEADBEEF));
```

Comment:
This is not safe. We need to sample random numbers to provide zero-knowledge. Please change these everywhere to call into a secure random number generator.